packages/fetchye/__tests__/computeKey.spec.js

```diff
@@ -58,4 +67,55 @@ describe('computeKey', () => {
     };
     expect(computeKey('uri', firstOptions).hash).toBe(computeKey('uri', secondOptions).hash);
   });
+
+  it('should return a different, stable hash, if the option mapKeyToCacheKey is passed', () => {
+    expect(computeKey(() => 'abcd', {
+      mapKeyToCacheKey: () => 'efgh',
+    })).toMatchInlineSnapshot(`
+      Object {
+        "hash": "a0e09d568bb5b47c046b0fac7a61ca10196151cc",
+        "key": "abcd",
+      }
+    `);
+  });
+
+  it('should pass generated cacheKey to the underlying hash function along with the options, and return the un-mapped key to the caller', () => {
+    const computedKey = computeKey(() => 'abcd', {
+      mapKeyToCacheKey: () => 'efgh',
+      optionKeyMock: 'optionKeyValue',
+    });
+    expect(computedKey.key).toBe('abcd');
+    expect(computeHash).toHaveBeenCalledWith(['efgh', { optionKeyMock: 'optionKeyValue' }], { respectType: false });
+  });
```

Comment:
```suggestion
  it('should pass generated cacheKey to the underlying hash function along with the options, and return the un-mapped key to the caller', () => {
    const computedKey = computeKey(() => 'abcd', {
      mapKeyToCacheKey: (key, options => `${key.toUpperCase()}-${options.optionKeyMock}`,
      optionKeyMock: 'optionKeyValue',
    });
    expect(computedKey.key).toBe('abcd');
    expect(computeHash).toHaveBeenCalledWith(['ABCD-optionKeyValue', { optionKeyMock: 'optionKeyValue' }], { respectType: false });
  });
```

Reply:
The option key value is already passed to the hash function. All the options except the two mapping functions are. there would be no value in including the option in the cacheKey.

I do like the idea of transforming the key in the tests though, will include that change.

Reply:
The idea isn't that the option is needed in the cache key, but it validates the mapping function was called with that. Otherwise you could use a mock function